### PR TITLE
test(docs-infra): ensure RxJS-related docs examples can compile without errors

### PR DESCRIPTION
**Pull request by sonukapoor**
Some examples are simple typescript files. The changes in this PR make sure that they compile.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #28017 

## What is the new behavior?
There is currently no way to validate code that is not an angular application e.g: The RxJS guide has no tests and no compilation steps. This PR adds a compilation step to at least validate the files.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by sonukapoor** [comment]
@googlebot I fixed it.

File: aio/content/examples/observables-in-angular/example-config.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+      {
+        "cmd": "yarn",
+        "args": [
+          "tsc",
+          "--project",
+          "./"
+        ]
+      }
+    ]
+  }
```
Comment (gkalpak) [inline aio/content/examples/observables-in-angular/example-config.json]: Please, fix the indentation (in this and other similar files below).

File: aio/content/examples/observables-in-angular/src/main.ts [hunk 2]
```diff
@@ -39,9 +39,9 @@ export class ZippyComponent {
        Time: {{ time | async }}</div>`
 })
 export class AsyncObservablePipeComponent {
-  time = new Observable(observer =>
-    setInterval(() => observer.next(new Date().toString()), 1000)
-  );
+  time: Observable<string> = new Observable(observer => {
```
Comment (gkalpak) [inline aio/content/examples/observables-in-angular/src/main.ts]: Why not use `time = new Observable<string>(...)`?

File: aio/content/examples/observables-in-angular/example-config.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+    "e2e": [
+      {
+        "cmd": "yarn",
+        "args": [
+          "tsc",
+          "--project",
+          "./"
+        ]
```
Comment (gkalpak) [inline aio/content/examples/observables-in-angular/example-config.json]: Nit: Since this command is short, I think it would be more readable as:

```json
{"cmd": "yarn", "args": ["tsc", "--project", "./"]}
```

or

```json
{
  "cmd": "yarn",
  "args": ["tsc", "--project", "./"]
}
```

File: aio/package.json [hunk 4]
```diff
@@ -112,7 +112,7 @@
     "@types/jasmine": "^3.4.2",
     "@types/jasminewd2": "^2.0.8",
     "@types/lunr": "^2.3.2",
-    "@types/node": "^12.7.9",
+    "@types/node": "^12.12.14",
```
Comment (gkalpak) [inline aio/package.json]: OOC, why this change? (It looks unrelated.)

File: aio/content/examples/practical-observable-usage/src/typeahead.ts [hunk 5]
```diff
@@ -1,12 +1,12 @@
-
 import { fromEvent } from 'rxjs';
 import { ajax } from 'rxjs/ajax';
-import { map, filter, debounceTime, distinctUntilChanged, switchMap } from 'rxjs/operators';
+import { debounceTime, distinctUntilChanged, filter, map, switchMap } from 'rxjs/operators';
```
Comment (gkalpak) [inline aio/content/examples/practical-observable-usage/src/typeahead.ts]: ❤️

**Review by gkalpak — CHANGES_REQUESTED** [review]
Thx, @sonukapoor! This is looking great already. I left a couple of comments.
Also, can you please update the commit message to something like:

> test(docs-infra): ensure RxJS-related docs examples can compile without errors
>
> Previously, some RxJS-related examples (which are not proper Angular apps) were not
> tested on CI as part of the `example-e2e` npm script. This meant that the examples
> could get out-of-date or contain compile errors without as noticing.
>
> This commit ensures that the `example-e2e` script picks up these examples and checks
> that they compile successfully.
>
> Partly addresses #28017.

<sub>This should _not_ close #28017, because that one is about adding tests (which this PR does not do).</sub>

File: aio/package.json (hunk 4 above)
Comment (sonukapoor) [reply aio/package.json]: CI is failing with the following error:

`../../../tools/examples/shared/node_modules/typescript/lib/lib.es2015.iterable.d.ts(41,6): error TS2300: Duplicate identifier 'IteratorResult'.`

According to the the typescript repo an update is required to the @types/node package:

https://github.com/microsoft/TypeScript/issues/32333

So, it's kind of related to the work in this PR. What do you think @gkalpak ?

**Comment by sonukapoor** [comment]
@gkalpak All changes done, except the package.json one. Please suggest what should be done on that part.

File: aio/content/examples/observables/example-config.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+    "e2e": [
+      {
+        "cmd": "yarn",
+        "args": [ "tsc", "--project", "./" ]
+      }
+    ]
+  }
```
Comment (gkalpak) [inline aio/content/examples/observables/example-config.json]: Please, fix the indentation in this file.

File: aio/package.json (hunk 4 above)
Comment (gkalpak) [reply aio/package.json]: Actually, the examples use [aio/tools/examples/shared/package.json][1].
You should revert this change and update `@types/node` there?

[1]: https://github.com/angular/angular/blob/168abc6d6/aio/tools/examples/shared/package.json

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/content/examples/observables/example-config.json (hunk 6 above)
Comment (sonukapoor) [reply aio/content/examples/observables/example-config.json]: Fixed. I have no idea how you saw that. :)

**Comment by sonukapoor** [comment]
@gkalpak I have reverted the @types/node in root and add it to the shared package.json file. I also had to upgrade @types/jquery to 3.3.28. Please review.

File: aio/package.json (hunk 4 above)
Comment (sonukapoor) [reply aio/package.json]: Done.

File: aio/yarn.lock [hunk 7]
```diff
@@ -1242,9 +1242,9 @@
   integrity sha512-wp6IOGu1lxsfnrD+5mX6qwSwWuqsdkKKxTN4aQc4wByHAKZJf9/D4KXPQ1POUjEbnCP5LMggB0OEFNY9OTsMqg==
 
 "@types/node@^12.7.9":
-  version "12.7.9"
-  resolved "https://registry.yarnpkg.com/@types/node/-/node-12.7.9.tgz#da0210f91096aa67138cf5afd04c4d629f8a406a"
-  integrity sha512-P57oKTJ/vYivL2BCfxCC5tQjlS8qW31pbOL6qt99Yrjm95YdHgNZwjrTTjMBh+C2/y6PXIX4oz253+jUzxKKfQ==
+  version "12.12.14"
+  resolved "https://registry.yarnpkg.com/@types/node/-/node-12.12.14.tgz#1c1d6e3c75dba466e0326948d56e8bd72a1903d2"
+  integrity sha512-u/SJDyXwuihpwjXy7hOOghagLEV1KdAST6syfnOk6QZAMzZuWZqXy5aYYZbh8Jdpd4escVFP0MvftHNDb9pruA==
```
Comment (gkalpak) [inline aio/yarn.lock]: Please revert this change, since it is not needed any more.

**Review by gkalpak — APPROVED** [review]
Great job, @sonukapoor 👏
Can you also please squash all commits together (and keep the commit message from https://github.com/angular/angular/commit/a610a01be)?

**Comment by gkalpak** [comment]
@googlebot I consent.

**Comment by gkalpak** [comment]
**merge-assistance:** Global approval for this docs(-infra)-only change.
